odoo/addons/base/res/res_lang.py

```diff
@@ -239,6 +239,14 @@ def unlink(self):
             self.env['ir.translation'].search([('lang', '=', language.code)]).unlink()
         self.clear_caches()
         return super(Lang, self).unlink()
+    
+    @api.model
+    def context_lang_format(self, value):
+        """Format the Date as of in Language Settings (based on 'lang' in context)"""
+        if isinstance(value, basestring):
+            value = fields.Datetime.from_string(value)
+        lang = self._lang_get(self.env.context.get('lang', 'en_US'))
+        return fields.datetime.strftime(value, lang.date_format)
```

Comment:
strftime being a piece of crap, I'd rather we used babel.

Reply:
I cannot see any function for formatting dates according to the Date Format set on Settings/Language Settings.
Eg: Default date format is '%Y-%M-%D'. I set Date Format as '%D/%M/%Y'. Then everywhere in odoo the date format changes. What if i need to change the dateformat in code.? Like using the context['lang'].

Reply:
> I cannot see any function for formatting dates according to the Date Format set on Settings/Language Settings.

Seems to be what [format_date](https://github.com/odoo/odoo/blob/c201cf2b77cf44e9ad15165ceadde41dfaa20ceb/odoo/tools/misc.py#L1095) does unless you pass in an explicit format?

Reply:
Sorry, it seems odoo-11 have it. mistook odoo-11 for odoo-10. format_date not in odoo10, tools.misc

Reply:
New features don't generally get added/backported to previous versions, if format_date was not  backported this new equivalent to it probably would not be either, I fear.